awx/api/serializers.py

```diff
@@ -4419,6 +4555,333 @@ def validate(self, attrs):
+            else:
+                node['identifier'] = str(uuid4())
+
+        requested_ujts = {j['unified_job_template'] for j in attrs['jobs']}
+        requested_use_inventories = {job['inventory'] for job in attrs['jobs'] if 'inventory' in job}
+        requested_use_execution_environments = {job['execution_environment'] for job in attrs['jobs'] if 'execution_environment' in job}
+        requested_use_credentials = set()
+        requested_use_labels = set()
+        requested_use_instance_groups = set()
+        for job in attrs['jobs']:
+            if 'credentials' in job:
+                [requested_use_credentials.add(cred) for cred in job['credentials']]
+            if 'labels' in job:
+                [requested_use_labels.add(label) for label in job['labels']]
+            if 'instance_groups' in job:
+                [requested_use_instance_groups.add(instance_group) for instance_group in job['instance_groups']]
+
+        # If we are not a superuser, check we have permissions
+        # TODO: As we add other related items, we need to add them here
+        if request and not request.user.is_superuser:
+            self.check_organization_permission(attrs, request)
+            self.check_unified_job_permission(request, requested_ujts)
+            if requested_use_inventories:
+                self.check_inventory_permission(request, requested_use_inventories)
+
+            if requested_use_credentials:
+                self.check_credential_permission(request, requested_use_credentials)
+
+            if requested_use_labels:
+                self.check_label_permission(requested_use_labels)
+
+            if requested_use_instance_groups:
+                self.check_instance_group_permission(request, requested_use_instance_groups)
+
+            if requested_use_execution_environments:
+                self.check_instance_group_permission(request, requested_use_instance_groups)
+
+        # all of the unified job templates and related items have now been checked, we can now grab the objects from the DB
+        jobs_object = self.get_objectified_jobs(
+            attrs,
+            requested_ujts,
+            requested_use_inventories,
+            requested_use_credentials,
+            requested_use_labels,
+            requested_use_instance_groups,
+            requested_use_execution_environments,
+        )
+
+        attrs['jobs'] = jobs_object
+        if 'extra_vars' in attrs:
+            extra_vars_dict = parse_yaml_or_json(attrs['extra_vars'])
+            attrs['extra_vars'] = json.dumps(extra_vars_dict)
+        attrs = super().validate(attrs)
+        return attrs
+
+    def create(self, validated_data):
+        job_node_data = validated_data.pop('jobs')
+        # FIXME: Need to set organization on the WorkflowJob in order for users to be able to see it --
+        # normally their permission is sourced from the underlying WorkflowJobTemplate
+        # maybe we need to add Organization to WorkflowJob
```

Comment:
WorkflowJob does have an organization, and you have already added it to this serializer. So delete comment, as work has been finished?

Reply:
removed